Makefile

```diff
@@ -10,19 +10,46 @@ GOHOSTOS=$(shell go env GOHOSTOS)
 GOHOSTARCH=$(shell go env GOHOSTARCH)
 GO_MOD_VERSION=$(shell grep "^go" go.mod | awk '{print $$2}')
 GO_INSTALLED_VERSION=$(shell go version | awk '{print $$3}' | sed -e /.*go/s///)
-export CGO_ENABLED=0
 
+# Build number passed in must be a monotonic int representing
+# the build.
+JUJU_BUILD_NUMBER ?=
+
+# JUJU_VERSION is the JUJU version currently being represented in this
+# repository.
+JUJU_VERSION=$(shell go run -ldflags "-X $(PROJECT)/version.build=$(JUJU_BUILD_NUMBER)" version/helper/main.go)
+
+# BUILD_DIR is the directory relative to this project where we place build
+# artifacts created
+# by this Makefile
 BUILD_DIR ?= $(PROJECT_DIR)/_build
+
 BIN_DIR = ${BUILD_DIR}/${GOOS}_${GOARCH}/bin
 
+# SIMPLESTREAMS_DIR is the directory where we place simple streams archives for
+# built juju binaries.
+SIMPLESTREAMS_DIR ?= ${BUILD_DIR}/simplestreams
+
 # TEST_PACKAGE_LIST is the path to a file that is a newline delimited list of
 # packages to test. This file must be sorted.
 TEST_PACKAGE_LIST ?=
 
+# Explicitly tell GO that we don't want CGO in our builds
+export CGO_ENABLED=0
+
+# bin_platform_path calculates the bin directory path for build artifacts for
+# the list of Go style platforms passed to this macro. For example
+# linux/amd64 linux/arm64
+bin_platform_path = $(addprefix ${BUILD_DIR}/, $(addsuffix /bin, $(subst /,_,${1})))
```

Comment:
Is there a "better" or different name for this? It's awfully close to "bin_platform_paths" below
eg simplestreams_bin_path

Actually, it only appears to be used in one place below so do we even need a macro, can we just use the code directly in the one place it is called? Or at the least, move this closer to where it is used? Prefer just ditching it.